lib/src/main/kotlin/com/lemonappdev/konsist/core/provider/KoFullyQualifiedNameProviderCore.kt

```diff
@@ -1,5 +1,42 @@
 package com.lemonappdev.konsist.core.provider
 
+import com.lemonappdev.konsist.api.declaration.KoAnnotationDeclaration
 import com.lemonappdev.konsist.api.provider.KoFullyQualifiedNameProvider
+import com.lemonappdev.konsist.api.provider.KoNullableTypeProvider
 
-internal interface KoFullyQualifiedNameProviderCore : KoFullyQualifiedNameProvider, KoBaseProviderCore
+internal interface KoFullyQualifiedNameProviderCore :
+    KoFullyQualifiedNameProvider,
+    KoBaseProviderCore,
+    KoContainingFileProviderCore,
+    KoNameProviderCore {
+    val textUsedToFqn: String
+        get() = name
+
+    override val fullyQualifiedName: String
+        get() {
+            val isInImports = containingFile
```

Comment:
this variable stores the name, so it should be called `nameFromImports`.

Also logic should be updated:
1. get name from imports 
2. if name is null then get name from file
3. return name